File: m2-h2/src/main/java/ru/yegorpilipenko/otus/spring/m2h2/service/AuthorService.java
package ru.yegorpilipenko.otus.spring.m2h2.service;

import ru.yegorpilipenko.otus.spring.m2h2.document.Author;

import javax.annotation.Nonnull;
import java.util.Collection;

public interface AuthorService {

    @Nonnull
    Collection<Author> getAll();

}

File: m1-h6/src/main/kotlin/ru/yegorpilipenko/otus/spring/m1h6/service/AuthorService.kt
package ru.yegorpilipenko.otus.spring.m1h6.service

import ru.yegorpilipenko.otus.spring.m1h6.entity.Author

interface AuthorService {

    fun findAll(): Collection<Author>

    fun findById(id: Long): Author?

    fun save(author: Author): Author

    fun removeById(id: Long): Boolean
}